### ISSUE TITLE
Passphrase is asked repeatedly for T1

### ISSUE THREAD

**Issue by tsusanka** [trezor/trezor-suite#1272]
In the newest RC firmwares as sent to QA we have introduced the capability of caching up to 10 simultaneous sessions (=passphrases) in Trezor. That means the user needs to enter the passphrase only once, Suite remembers the session id and when sent correctly the passphrase is not prompted again.

As @bosomt reported this works fine with TT 2.3.0 but does not with T1 1.9.0. 

**How to reproduce**
1. Enable passphrase.
2. Create new hidden wallet, use passphrase "a".
3. Create another hidden wallet, use passphrase "b".
4. Go back to the hidden wallet created in (2).
5. Ask for a new address.
6. The passphrase is prompted but should not be!

**Hint**
It seems T1 is receiving `Initialize.session_id = null` after changing from one hidden wallet to another but it should receive the previously stored session id.

**Revision**
I have tested this on trezor-firmware's 624b63, to get the firmware go to https://gitlab.com/satoshilabs/trezor/trezor-firmware/pipelines/122708388 and download either "legacy emu regular build" for emulator or "legacy fw regular build". **This will not work on the latest publicly released 2.2.0 and 1.8.3 because it doesn't have the new passphrase redesign.**

**Comment by szymonlesisz** [comment]
should be fixed by:
https://github.com/trezor/connect/commit/2beca2599c524ff9a5c8dd9196c901c6542f1aad

**Comment by bosomt** [comment]
was it merged already to master suite branch ?
or is there special branch ?
im using less than 10 passphrases but doesn't remember them

**Comment by sorooris** [comment]
https://suite.corp.sldev.cz/suite-web/develop/
Chromium (WebUSB connection)

Issue **still persists** on T1 and TT (1.9.0 & 2.3.0, revision **86b010a1**):
Selecting one of the previously created nine hidden wallets (with passphrase) does incorrectly require passphrase entry.

**Comment by tsusanka** [comment]
It turns out this is not a bug. It is more of an incorrect test scenario description from my side.

Trezor can hold up to ten _sessions_, however sessions are not passphrases. In particular, Suite needs to send `Initialize(session_id=null)` anytime it needs to invoke the `PassphraseRequest`. That means that already at the beginning, Suite sends two Initialize messages and _spoils_ two spots. Also when you create a new hidden wallet, you need to confirm the passphrase again (if the wallet is empty), which is done by sending `Initialize(session_id=null)` again, effectively spoiling two spots for one passphrase.

In a nutshell, we can't guarantee the number of possible simultaneous passphrases. In 90% it should be 5 passphrases when creating new ones and it should be 10 when you are only switching between the passphrases and not creating new ones. However it depends on the concrete behaviour in Suite.

---

@bosomt @sorooris could you please retest and confirm this works for let's say three passphrases? 

And I believe we can close after that.

**Comment by bosomt** [comment]
@tsusanka it works fine for me, you can close it

**Comment by sorooris** [comment]
Couldn't replicate, QA OK.

**Comment by bosomt** [comment]
works fine now
